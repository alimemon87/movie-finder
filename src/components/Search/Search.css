.Search{
    width :100%;
    margin:auto;
    background: #f4f4f4;
    padding: 0 20px 20px 20px;
    border: 1px #ccc solid;
    border-radius: 10px; 
}

.textCenter{
    text-align: center;
}