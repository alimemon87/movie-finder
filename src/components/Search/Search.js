import React, {useState, Fragment} from "react";
//import classes from './Search.css';

import {FormControl, Button} from 'react-bootstrap';

const Search = (props) => {
    //Using Hook
    const [result,setResult] = useState("");


    const inputHandler = (e) => {
        //e.preventDefault();
        setResult(e.target.value);
    }

    //Make all fields empty
    const emptyFields = () => {
        setResult("");
    }

    //When search button pressed
    const clickHandler = (e) => {
        e.preventDefault();
        props.search(result);
        emptyFields();
    }

    return(
        
        <Fragment>
            <FormControl type="text" placeholder="Search" value={result} onChange={inputHandler} className="mr-sm-2" />
            <Button type="submit" id="my-button" variant="outline-success" onClick={clickHandler}>Search</Button>
        </Fragment>

        
        
    );
}

export default Search;