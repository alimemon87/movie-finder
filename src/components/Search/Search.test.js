import React from 'react'                                              
import {shallow} from 'enzyme';        
import renderer from 'react-test-renderer';
import Search from './Search';

it('matches the snapshot', () => {
  const tree = renderer.create(<Search />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('SearchForm', () => {
    it('check button', () => {
      // Mock search form parameters.
      const searchQuery = 'kittens';
      const onSubmit = jest.fn();
  
      // Create test component instance.
      const testComponentInstance = renderer.create((
        <Search query={searchQuery} onSearchSubmit={onSubmit} />
      )).root;
  
      // Try to find submit button inside the form.
      const submitButtonInstance = testComponentInstance.findByProps({
        type: 'submit',
      });
      expect(submitButtonInstance).toBeDefined();
  
    });
  });

  describe('Seach', () => {
    it('button click should hide component', () => {
      const component = shallow(<Search />);
      const wrapper = component.find('#my-button');
      expect(wrapper.length).toBe(1);
    });
  });

  describe('Searchbar', () => {
    it('Check search field', () => {
      const component = shallow(<Search />);
      const wrapper = component.find('.mr-sm-2');
      expect(wrapper.length).toBe(1);
    }); 
  });

  
 