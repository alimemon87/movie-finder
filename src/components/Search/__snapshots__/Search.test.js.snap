// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
Array [
  <input
    className="mr-sm-2 form-control"
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />,
  <button
    className="btn btn-outline-success"
    disabled={false}
    id="my-button"
    onClick={[Function]}
    type="submit"
  >
    Search
  </button>,
]
`;
